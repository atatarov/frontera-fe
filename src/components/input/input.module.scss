.FormItem {
  position: relative;
  margin: 0;
  padding: 0;
}

.FormItem input {
  border: 0.5px solid var(--gray-light);
}

.FormItem:first-child input {
  border-radius: 8px 8px 0px 0px;
}

.FormItem:last-child input {
  border-radius: 0px 0px 8px 8px;
  border-top: none;

  margin-bottom: var(--offset-large);
}

.FormItem:not(:first-child) input {
  border-top: none;
}

.FormItem:only-child input {
  border-radius: 8px 8px 8px 8px;
}

.FormLabel {
  position: absolute;
  top: 30px;
  left: 24px;
}

.FormText:focus + .FormLabel {
  top: 8px;
  transition: top 0.1s ease-out;
}

.FormText::placeholder {
  color: transparent;
}

.FormText:focus::placeholder {
  color: var(--gray-light); 
}

.FormText:not(:focus):not(:placeholder-shown) + .FormLabel {
  top: 8px;
}

.FormLabel span {
  position: absolute;
  left: 24px;
  bottom: 24px;

  color: var(--purple-primary);

  font-weight: var(--font-weight-normal);
  font-size: var(--font-size);
  line-height: var(--line-height);
}

.FormText {
  box-sizing: border-box;
  height: 73px;
  width: 100%;
  padding: var(--offset) 0 0 var(--offset);

  font: inherit;
  background-color: transparent;

  border: none;
  outline: none;
}

.FormText:focus {
  background-color: var(--white);
  border: 0.5px solid var(--purple-primary);
  box-shadow: none;
}

@media (max-width: 532.98px) {
  .FormLabel input[placeholder], 
  .FormLabel span {
    font-size: var(--font-size);
    line-height: var(--line-height);
  }

  .FormItem:last-child input {
    margin-bottom: var(--offset);
  }
}