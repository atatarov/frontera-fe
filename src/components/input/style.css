
.form__label input[placeholder] {
  font-weight: normal;
  font-size: 18px;
  line-height: 27px;

  color: #541594;
}

.form__text {
  box-sizing: border-box;
  height: 73px;
  width: 100%;
  padding: 30px 0 0 30px;

  font: inherit;
  background-color: #C5B6D9;

  border: 0px solid #541594;
}

.form__text:focus {
  background-color: #f9f7ff;
}

.form__label {
  position: relative;
}

.form__label span {
  position: absolute;
  left: 30px;
  bottom: 30px;

  color: #541594;

  font-weight: normal;
  font-size: 18px;
  line-height: 27px;
}
