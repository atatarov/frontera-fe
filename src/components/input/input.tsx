import React, { HTMLAttributes } from 'react';
import PropTypes from 'prop-types';

import style from './style.css';

interface InputProps extends HTMLAttributes<HTMLInputElement> {
  label: string;
  id: string;
  name: string;
  type?:string;
  inputRef: React.RefObject<HTMLInputElement>;
}

export const Input: React.FC<InputProps> = ({ inputRef, label, id, name, type, ...rest }) => {
  return (
    <p className={style.formItem}>
      <label className={style.formLabel} htmlFor={String(id)}>
        <input
          ref={inputRef}
          className={style.formText}
          id={id}
          type={type}
          name={name}
          {...rest}
        />
        <span>{label}</span>
      </label>
    </p>
  );
};

Input.propTypes = {
  label: PropTypes.string.isRequired,
  id: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  type: PropTypes.oneOf(['email', 'password', 'text' ]),
}

Input.defaultProps = {
  type: 'text',
}