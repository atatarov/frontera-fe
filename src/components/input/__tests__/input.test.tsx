import React from "react";
import { mount } from 'enzyme';
import { describe, it, expect } from '@jest/globals';
import {Input} from "../input";
describe('Тестируем компонент input', () => {
    // it('Тестируем рендер компонента input', () => {
    //     const InputTest = mount(<Input {...props}/>);
    //     expect(InputTest).toMatchSnapshot();
    // });
    // it('Тестируем значение label', () => {
    //     const InputTest = mount(<Input {...props} />);
    //     expect(InputTest.prop('label')).toEqual('input');
    // });
    // it('Тестируем тип id', () => {
    //     const InputTest = mount(<Input {...props} />);
    //     expect(typeof InputTest.prop('id')).toBe('string');
    // });
    // it('Тестируем на null label', () => {
    //     const InputTest = mount(<Input {...props} />);
    //     expect(typeof InputTest.prop('id')).not.toBe(null);
    // });
    it.only('Тестируем событие', () => {
        const props1 = {
            id: '3',
            name: 'testInput',
            value: "test1",
            onChange: (v) => console.log(v)
        };
        const component = mount(<Input {...props1} />);
        expect(component).toMatchSnapshot();
        component.find('input').simulate('change', {
            target: { value: 'test2' }
        });
        component.update();
        expect(component).toMatchSnapshot();
        component.find('input').simulate('input', {
            target: { value: 'test3' }
        });
        component.update();
        expect(component).toMatchSnapshot();
    })
});