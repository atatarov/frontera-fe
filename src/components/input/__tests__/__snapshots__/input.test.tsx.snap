// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Тестируем компонент input Тестируем рендер компонента input 1`] = `
<Input
  label="label"
  type="text"
>
  <div
    className="formItem"
  >
    <label
      className="formLabel"
      htmlFor="undefined"
    >
      label
    </label>
    <input
      className="formText"
      type="text"
    />
  </div>
</Input>
`;

exports[`Тестируем компонент input Тестируем событие 1`] = `
<Input
  name="value"
  onChange={[Function]}
  type="text"
  value="test1"
>
  <div
    className="formItem"
  >
    <input
      className="formText"
      name="value"
      onChange={[Function]}
      type="text"
      value="test1"
    />
  </div>
</Input>
`;

exports[`Тестируем компонент input Тестируем событие 2`] = `
<Input
  name="value"
  onChange={[Function]}
  type="text"
  value="test2"
>
  <div
    className="formItem"
  >
    <input
      className="formText"
      name="value"
      onChange={[Function]}
      type="text"
      value="test2"
    />
  </div>
</Input>
`;
